versions/1.1/SPEC.md

```diff
@@ -0,0 +1,5804 @@
+
+### Task Inputs
+
+A task's `input` section declares its input parameters. The values for declarations within the input section may be specified by the caller of the task. An input declaration may be initialized to a default value or expression that will be used when the caller does not specify a value. Input declarations may also be optional, in which case a value may be specified but is not required. If an input declaration is not optional and does not have an initialization, then it is a required input, meaning the caller must specify a value.
+
+```wdl
+task t {
+  input {
+    Int i               # a required input parameter
+    String s = "hello"  # an input parameter with a default value
+    File? f             # an optional input parameter
+  }
+
+  # [... other task sections]
+}
+```
+
+#### Task Input Localization
+
+`File` inputs must be treated specially since they may require localization to the execution directory. For example, a file located on a remote web server that is provided to the execution engine as an `https://` URL must first be downloaded to the machine where the task is being executed.
+
+- Files are localized into the execution directory prior to the task execution commencing.
+- When localizing a `File`, the engine may choose to place the file wherever it likes so long as it adheres to these rules:
+  - The original file name must be preserved even if the path to it has changed.
+  - Two input files with the same name must be located separately, to avoid name collision.
+  - Two input files that originated in the same storage directory must also be localized into the same directory for task execution (see the special case handling for Versioning Filesystems below).
+- When a WDL author uses a `File` input in their [Command Section](#command-section), the fully qualified, localized path to the file is substituted when that declaration is referenced in the command template.
+
+##### Special Case: Versioning Filesystem
+
+Two or more versions of a file in a versioning filesystem might have the same name and come from the same directory. In that case, the following special procedure must be used to avoid collision:
+  - The first file is always placed according to the rules above.
+  - Subsequent files that would otherwise overwrite this file are instead placed in a subdirectory named for the version.
+
+For example, imagine two versions of file `fs://path/to/A.txt` are being localized (labeled version `1.0` and `1.1`). The first might be localized as `/execution_dir/path/to/A.txt`. The second must then be placed in `/execution_dir/path/to/1.1/A.txt`
+
+#### Input Type Constraints
+
+Recall that a type may have a quantifier:
+* `?` means that the parameter is optional. A user does not need to specify a value for the parameter in order to satisfy all the inputs to the workflow.
+* `+` applies only to `Array` types and it represents a constraint that the `Array` value must contain one-or-more elements.
```

Comment:
This is far more clear/succinct than I'd seen it before.... or wait, did this ever get mentioned before?  These two bullets need to be in multiple places perhaps.  Or not here, in the other places where they are mentioned.